cognite/pygen/_query/interface.py

```diff
@@ -61,24 +65,56 @@ def search(
         Returns:
             list[dict[str, Any]]: The search results.
 
+        Raises:
+            ValueError: If the view is not an edge view and nested properties are used, e.g. {"property": ["nested"]}.
+            CogniteAPIError: If the view is not found.
+
         """
         filter = self._equals_none_to_not_exists(filter)
-        search_result = self._client.data_modeling.instances.search(
-            view,
+        return self._execute_search(view, properties, query, filter, search_properties, sort, limit)
+
+    def _execute_search(
+        self,
+        view_id: dm.ViewId,
+        properties: SelectedProperties | None = None,
+        query: str | None = None,
+        filter: filters.Filter | None = None,
+        search_properties: str | SequenceNotStr[str] | None = None,
+        sort: Sequence[dm.InstanceSort] | dm.InstanceSort | None = None,
+        limit: int | None = None,
+    ) -> list[dict[str, Any]]:
+        view = self._get_view(view_id)
+        flatten_props = self._as_property_list(properties, "list") if properties else None
+        are_flat_properties = flatten_props == properties
```

Comment:
This is a bit confusing to me. Maybe consider having a `is_flattened` validation function ?